File: app/src/main/java/com/renameExample/android/learnChinese/Word.java
package com.renameExample.android.learnChinese;

/**
 * {@link Word} represents a vocabulary word that the user wants to learn.
 * It contains a default translation and a Miwok translation for that word.
 */
public class Word {

    /** Default translation for the word */
    private String mDefaultTranslation;

    /** Miwok translation for the word */
    private String mMiwokTranslation;

    /** Image resource ID for the word */
    private int mImageResourceId = NO_IMAGE_PROVIDED;

    /** Constant vlaue that represents no image was provided for this word */
    private static final int NO_IMAGE_PROVIDED = -1;

    /** Audio resource ID for this word */
    private int mAudioResourceId;

    /**
     * Create a new Word object.
     *
     * @param defaultTranslation is the word in a language that the user is already familiar with
     *                           (such as English)
     * @param chineseTranslation is the word in the Miwok language
     *
     * @param audioResourceId is the resource ID for the audio file associated with this word
     */
    public Word(String defaultTranslation, String chineseTranslation, int audioResourceId) {
        mDefaultTranslation = defaultTranslation;
        mMiwokTranslation = chineseTranslation;
        mAudioResourceId = audioResourceId;
    }

    /**
     * Create a new Word object.
     *
     * @param defaultTranslation is the word in a language that the user is already familiar with
     *                           (such as English)
     * @param chineseTranslation is the word in the Miwok language
     */
    public Word(String defaultTranslation, String chineseTranslation, int imageResourceId, int audioResourceId) {
        mDefaultTranslation = defaultTranslation;
        mMiwokTranslation = chineseTranslation;
        mImageResourceId = imageResourceId;
        mAudioResourceId = audioResourceId;
    }

    /**
     * Get the default translation of the word.
     */
    public String getDefaultTranslation() {
        return mDefaultTranslation;
    }

    /**
     * Get the Miwok translation of the word.
     */
    public String getMiwokTranslation() {
        return mMiwokTranslation;
    }

    /**
     * Return the image resource ID of the word.
     */
    public int getmImageResourceId() {
        return mImageResourceId;
    }

    /**
     * Return whether or not there is an image for this word.
     * @return
     */
    public boolean hasImage() {
        return mImageResourceId != NO_IMAGE_PROVIDED;
    }

    /**
     * Return the sudio resource ID of this word.
     * @return
     */
    public int getmAudioResourceId() {
        return mAudioResourceId;
    }
}

File: app/src/main/java/com/renameExample/android/learnChinese/model/Word.kt
package com.renameExample.android.learnChinese.model

class Word {
    var defaultTranslation: String
        private set
    var miwokTranslation: String
        private set
    private var mImageResourceId = NO_IMAGE_PROVIDED
    private var mAudioResourceId: Int

    constructor(defaultTranslation: String, chineseTranslation: String, audioResourceId: Int) {
        this.defaultTranslation = defaultTranslation
        miwokTranslation = chineseTranslation
        mAudioResourceId = audioResourceId
    }

    constructor(
        defaultTranslation: String,
        chineseTranslation: String,
        imageResourceId: Int,
        audioResourceId: Int
    ) {
        this.defaultTranslation = defaultTranslation
        miwokTranslation = chineseTranslation
        mImageResourceId = imageResourceId
        mAudioResourceId = audioResourceId
    }

    fun getmImageResourceId(): Int {
        return mImageResourceId
    }

    fun hasImage(): Boolean {
        return mImageResourceId != NO_IMAGE_PROVIDED
    }

    fun getmAudioResourceId(): Int {
        return mAudioResourceId
    }

    companion object {
        private const val NO_IMAGE_PROVIDED = -1
    }
}
